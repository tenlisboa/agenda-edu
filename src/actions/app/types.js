export const SET_EVENTS = 'SET_EVENTS'
export const SELECT_EVENT = 'SELECT_EVENT'
export const BUSY = 'BUSY'
