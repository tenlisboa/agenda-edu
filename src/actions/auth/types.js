export const ERROR = 'ERROR'
export const BUSY = 'BUSY'
