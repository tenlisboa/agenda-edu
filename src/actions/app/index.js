import api from '../../utils/api'
import { AsyncStorage } from 'react-native'
import { SET_EVENTS } from './types'

export const fetchEvents = (page = 1) => async dispatch => {
  const token = await AsyncStorage.getItem('@Auth:Token')
  const { data } = await api.get(`/events?limit=10;page=${page}`, {
    headers: { token }
  })
  dispatch({
    type: SET_EVENTS,
    events: data
  })
}
