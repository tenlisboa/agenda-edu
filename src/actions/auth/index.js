import { ERROR, BUSY } from './types'
import api from '../../utils/api'
import { AsyncStorage } from 'react-native'

export const login = (email, password) => async dispatch => {
  dispatch({ type: BUSY, busy: true })
  try {
    const { data } = await api.post('/login', { email, password })
    await AsyncStorage.setItem('@Auth:Token', data.token)
    dispatch({ type: BUSY, busy: false })
    dispatch({ type: ERROR, error: null })
    return data.token
  } catch (err) {
    dispatch({ type: ERROR, error: err.response.data.message })
    dispatch({ type: BUSY, busy: false })
  }
}
