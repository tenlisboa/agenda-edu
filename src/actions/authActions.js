import { LOGIN } from './actionTypes'

export const login = (email, password) => {
  const token = 'aksdjfaoidsfj12308rfa0fn23fna'

  return {
    type: LOGIN,
    payload: token
  }
}
