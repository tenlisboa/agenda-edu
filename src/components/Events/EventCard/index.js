import React from 'react'
import { Text, TouchableOpacity } from 'react-native'
import { FontAwesome5 as Icon } from '@expo/vector-icons'
import {
  Card,
  Label,
  Title,
  TimeArea,
  Time,
  Date,
  Image,
  TextContainer,
  CardWrapper
} from './styles'
import { dateToLong, getHourAndMinutes } from '../../../utils/date'

const shadow = {
  shadowColor: '#000',
  shadowOpacity: 0.3,
  shadowRadius: 12,
  shadowOffset: { width: 0, height: 0 },
  elevation: 3
}

const EventCard = props => (
  <TouchableOpacity onPress={props.onPress}>
    <CardWrapper hasMarginBottom={props.hasMarginBottom} style={shadow}>
      <Card>
        <Image
          source={{
            uri: props.image
          }}
        />
        <TextContainer>
          <Label>{'Eventos'.toUpperCase()}</Label>
          <Title>{props.title}</Title>
          <TimeArea>
            <Text style={{ marginRight: 5 }}>
              <Icon name="clock" size={16} color="#666666" />
            </Text>
            <Time>{getHourAndMinutes(props.startAt)}</Time>
          </TimeArea>
          <Date>{dateToLong(props.sendAt, true)}</Date>
        </TextContainer>
      </Card>
    </CardWrapper>
  </TouchableOpacity>
)

export default EventCard
