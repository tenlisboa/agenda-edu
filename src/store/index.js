import { createStore, combineReducers, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'

import Reducers from '../reducers'

export const store = createStore(
  combineReducers(Reducers),
  applyMiddleware(thunk)
)
