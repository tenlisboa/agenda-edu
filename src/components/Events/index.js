import React, { Component } from 'react'
import { View, ActivityIndicator } from 'react-native'

import EventCard from './EventCard'

import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { fetchEvents } from '../../actions/app'

import Divider from '../Shared/Divider'
import AppContainer from '../AppContainer'
import { Screen, Loading } from './styles'

class Events extends Component {
  componentWillMount() {
    const { fetchEvents, metadata } = this.props
    if (!metadata.page) {
      fetchEvents()
    }
  }

  loadMore = () => {
    if (this.props.busy) return
    const {
      metadata: { page },
      fetchEvents
    } = this.props

    fetchEvents(parseInt(page) + 1)
  }

  openMenu = () => {
    alert('Open Menu')
  }

  renderFooter = () => {
    if (!this.props.busy) return null
    return (
      <Loading>
        <ActivityIndicator size="large" />
      </Loading>
    )
  }

  render() {
    const { events } = this.props
    return (
      <AppContainer
        showHeader
        title="Eventos"
        icon="bars"
        iconOnPress={this.openMenu}
      >
        <Screen
          showsVerticalScrollIndicator={false}
          contentContainerStyle={{ paddingBottom: 20 }}
          data={events}
          keyExtractor={item => item.data[0].id.toString()}
          onEndReached={this.loadMore}
          onEndReachedThreshold={0.1}
          ListFooterComponent={this.renderFooter}
          renderItem={({ item }) => (
            <View>
              <Divider text={item.date} />
              {item.data.map((event, index) => (
                <EventCard
                  key={event.id}
                  hasMarginBottom={index + 1 !== item.data.length}
                  {...event}
                />
              ))}
            </View>
          )}
        />
      </AppContainer>
    )
  }
}

const mapStateToProps = state => ({
  events: state.app.events,
  metadata: state.app.metadata,
  busy: state.app.busy
})

const mapDispatchToProps = dispatch =>
  bindActionCreators({ fetchEvents }, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Events)
