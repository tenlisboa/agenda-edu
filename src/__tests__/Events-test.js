import React from 'react'
import { Provider } from 'react-redux'
import Events from '../components/Events'
import { store } from '../store'

import * as actions from '../actions/app'
import * as types from '../actions/app/types'
import reducer from '../reducers/app/eventsReducer'

import renderer from 'react-test-renderer'

test('Events renders correctly', () => {
  const container = () => (
    <Provider store={store}>
      <Events />
    </Provider>
  )
  const tree = renderer.create(container()).toJSON()
  expect(tree).toMatchSnapshot()
})

test('action of fetch events populate the store correctly', () => {
  const initialState = {
    busy: false,
    events: [],
    metadata: {},
    selectedEvent: {}
  }

  const events = {
    data: [
      {
        id: 1,
        title: 'Event 1',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque pretium nulla non arcu aliquam rhoncus eu sed leo. Aenean cursus nibh sit amet fringilla sodales. Vestibulum faucibus venenatis tempor. Morbi placerat ac massa id ultricies.',
        sendAt: '2019-10-06T11:28:19.688Z',
        image:
          'https://s3-us-west-2.amazonaws.com/agendaedu-dev/schools/c5c1a933-cdef-4c9b-8a87-490f25c2538d/events/5380/attachments/1550866911-$1-original-poster-agendakids.jpeg',
        startAt: '2019-10-14T10:40:36.036Z',
        location: 'Fake Street, 1001 - Fortaleza CE'
      },
      {
        id: 2,
        title: 'Event 2',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque pretium nulla non arcu aliquam rhoncus eu sed leo. Aenean cursus nibh sit amet fringilla sodales. Vestibulum faucibus venenatis tempor. Morbi placerat ac massa id ultricies.',
        sendAt: '2019-03-20T04:40:51.395Z',
        image:
          'https://s3-us-west-2.amazonaws.com/agendaedu-dev/schools/c5c1a933-cdef-4c9b-8a87-490f25c2538d/events/5380/attachments/1550866911-$1-original-poster-agendakids.jpeg',
        startAt: '2019-04-19T17:07:13.514Z',
        location: 'Fake Street, 1001 - Fortaleza CE'
      }
    ],
    metadata: {
      page: '1',
      limit: '10',
      pre_page: null,
      next_page: '11',
      total: 100,
      total_pages: 10
    }
  }

  expect(
    reducer(initialState, {
      type: types.SET_EVENTS,
      events
    })
  ).toEqual({
    busy: false,
    events: events.data,
    metadata: events.metadata,
    selectedEvent: {}
  })
})

test('action should select an event', () => {
  const initialState = {
    busy: false,
    events: [
      {
        id: 1,
        title: 'Event 1',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque pretium nulla non arcu aliquam rhoncus eu sed leo. Aenean cursus nibh sit amet fringilla sodales. Vestibulum faucibus venenatis tempor. Morbi placerat ac massa id ultricies.',
        sendAt: '2019-10-06T11:28:19.688Z',
        image:
          'https://s3-us-west-2.amazonaws.com/agendaedu-dev/schools/c5c1a933-cdef-4c9b-8a87-490f25c2538d/events/5380/attachments/1550866911-$1-original-poster-agendakids.jpeg',
        startAt: '2019-10-14T10:40:36.036Z',
        location: 'Fake Street, 1001 - Fortaleza CE'
      },
      {
        id: 2,
        title: 'Event 2',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque pretium nulla non arcu aliquam rhoncus eu sed leo. Aenean cursus nibh sit amet fringilla sodales. Vestibulum faucibus venenatis tempor. Morbi placerat ac massa id ultricies.',
        sendAt: '2019-03-20T04:40:51.395Z',
        image:
          'https://s3-us-west-2.amazonaws.com/agendaedu-dev/schools/c5c1a933-cdef-4c9b-8a87-490f25c2538d/events/5380/attachments/1550866911-$1-original-poster-agendakids.jpeg',
        startAt: '2019-04-19T17:07:13.514Z',
        location: 'Fake Street, 1001 - Fortaleza CE'
      }
    ],
    metadata: {},
    selectedEvent: {}
  }

  expect(
    reducer(initialState, {
      type: types.SELECT_EVENT,
      event: {
        id: 2,
        title: 'Event 2',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque pretium nulla non arcu aliquam rhoncus eu sed leo. Aenean cursus nibh sit amet fringilla sodales. Vestibulum faucibus venenatis tempor. Morbi placerat ac massa id ultricies.',
        sendAt: '2019-03-20T04:40:51.395Z',
        image:
          'https://s3-us-west-2.amazonaws.com/agendaedu-dev/schools/c5c1a933-cdef-4c9b-8a87-490f25c2538d/events/5380/attachments/1550866911-$1-original-poster-agendakids.jpeg',
        startAt: '2019-04-19T17:07:13.514Z',
        location: 'Fake Street, 1001 - Fortaleza CE'
      }
    })
  ).toEqual({
    busy: false,
    events: initialState.events,
    metadata: {},
    selectedEvent: {
      id: 2,
      title: 'Event 2',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque pretium nulla non arcu aliquam rhoncus eu sed leo. Aenean cursus nibh sit amet fringilla sodales. Vestibulum faucibus venenatis tempor. Morbi placerat ac massa id ultricies.',
      sendAt: '2019-03-20T04:40:51.395Z',
      image:
        'https://s3-us-west-2.amazonaws.com/agendaedu-dev/schools/c5c1a933-cdef-4c9b-8a87-490f25c2538d/events/5380/attachments/1550866911-$1-original-poster-agendakids.jpeg',
      startAt: '2019-04-19T17:07:13.514Z',
      location: 'Fake Street, 1001 - Fortaleza CE'
    }
  })
})
