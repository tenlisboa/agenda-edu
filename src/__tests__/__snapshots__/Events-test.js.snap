// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "borderWidth": 0,
          "left": 0,
          "position": "absolute",
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginTop": 20,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomWidth": 0.5,
              "borderColor": "#333333",
              "flexDirection": "row",
              "left": 0,
              "paddingBottom": 15,
              "paddingLeft": 21,
              "paddingRight": 21,
              "paddingTop": 15,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
                "marginLeft": 26,
              },
            ]
          }
        >
          Eventos
        </Text>
      </View>
    </View>
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.8)",
            "elevation": 0,
            "opacity": 0,
          },
        ]
      }
    />
  </View>
  <View
    elevation={0}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "borderWidth": 0,
          "elevation": 5,
          "left": -600,
          "position": "absolute",
          "shadowColor": "#000000",
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 600,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginTop": 20,
            "paddingLeft": 25,
            "paddingTop": 15,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "fontSize": 25,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Menu
      </Text>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#333333",
              "borderBottomWidth": 0.5,
              "marginBottom": 15,
              "marginTop": 15,
            },
          ]
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Text />
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontSize": 25,
                "fontWeight": "normal",
                "marginLeft": 10,
              },
            ]
          }
        >
          Sair
        </Text>
      </View>
    </View>
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
